### PR TITLE
ADD dax module

### PR DESCRIPTION
**Pull request by fastboot**
## Description

Description of what this PR does. What have you added or changed, and why?  If it fixes a bug or resolves a feature request, be sure to link to that issue.

**This PR adds a dax module to the already present terraform modules.**

## Review Checks

Please check if the PR fulfills these requirements:

_Put an `x` in the boxes that apply, Remove any lines that do not apply_
- [x] 📝 The commit message is clear and descriptive
- [x] 🔐 The Security Considerations section in the PR description is complete - **Please do not remove this**
- [ ] ✅ Tests for the changes have been added and run successfully including the new changes
- [x] 📄 Documentation has been added / updated (for bug fixes / features)

## Dependencies

Add links to any pull requests or documentation related to this pull request.

**None**

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- [ ] Yes
- [x] No breaking changes

## Security Considerations

Are there any other security or data concerns to be aware of?

Please discuss the security implications/considerations relevant to the proposed change. 
This may include...
* security-relevant design decisions
* concerns 
* important discussions
* implementation-specific guidance and pitfalls
* an outline of threats and risks and how they are being addressed.

**All security considerations in place.**

## Types of change

What kind of change does this Pull Request introduce?
_Put an `x` in the boxes that apply_

- [ ] 🐛 Bugfix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🛠 Adding or updating configuration files, development scripts etc.
- [ ] ♻️ Refactoring (no functional changes, no api changes)
- [ ] 🧹 Chore (removing redunant files, fixing typos etc.)
- [ ] 📄 Documentation Update 
- [ ] ❓ Other (if none of the other choices apply)

# Testing

Please include steps that the reviewer can follow in order to test the changes

1. Try to import and use dax module as explained in Readme for the repository.

### CODE REVIEW

File: dax/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| item\_cache\_ttl\_milli\_second | Time after which item cache will invalidate. Default 5 minutes | `number` | `300000` | no |
+| name | n/a | `string` | n/a | yes |
+| node\_type | n/a | `string` | `"dax.t2.small"` | no |
+| query\_cache\_ttl\_\_milli\_second | Time after which item query will invalidate. Default 5 minutes | `number` | `300000` | no |
+| region | n/a | `string` | n/a | yes |
+| replication\_factor | n/a | `number` | `3` | no |
+| stage | n/a | `string` | n/a | yes |
```
Comment (azhar22k) [inline dax/README.md]: for a generic reusable module, stage shouldn't be required

File: dax/main.tf [hunk 2]
```diff
@@ -0,0 +1,102 @@
+provider "aws" {
+  region = var.region
+  default_tags {
+    tags = var.tags
+  }
+}
+
+data "aws_caller_identity" "current" {}
+resource "aws_iam_role" "role" {
+  name = "${var.stage}-${var.name}-dax"
```
Comment (azhar22k) [inline dax/main.tf]: Same as https://github.com/peak-ai/terraform-modules/pull/13/files#r765448902
The user of the module should pass in the STAGE
Just think of any AWS resource, they don't force you to add stage to the resources

File: dax/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+output "endpoint_created" {
+  value       = aws_dax_cluster.cluster.cluster_address
+  description = "Endpoint of the created cluster."
```
Comment (azhar22k) [inline dax/outputs.tf]:
```suggestion
  description = "Endpoint of the DaX cluster."
```

File: dax/variables.tf [hunk 4]
```diff
@@ -0,0 +1,48 @@
+variable "stage" {
```
Comment (azhar22k) [inline dax/variables.tf]: same as https://github.com/peak-ai/terraform-modules/pull/13/files#r765448902

**Review by azhar22k — CHANGES_REQUESTED** [review]
Mostly looks good, just some comments about it being generic

File: dax/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+| tables | n/a | `list(string)` | n/a | yes |
+| tags | n/a | `any` | n/a | yes |
+| vpc\_id | n/a | `string` | n/a | yes |
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| endpoint\_created | Endpoint of the created cluster. |
```
Comment (azhar22k) [inline dax/README.md:29]: Can you add a Example use Just like the Other [modules](https://github.com/peak-ai/terraform-modules/tree/master/s3#example-usage)

Comment (fastboot) [reply dax/README.md:29]: Added!

File: dax/outputs.tf (hunk 3 above)
Comment (fastboot) [reply dax/outputs.tf]: Done

**Review by shreyance-jain — COMMENTED** [review]
looks fine, check for the other suggested changes.

File: dax/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
+## Requirements
+
+No requirements.
+
+## Providers
+
+| Name | Version |
+|------|---------|
+| aws | n/a |
```
Comment (shubhamPeak) [inline dax/README.md]: aws version need to mentioned.

File: dax/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+|------|-------------|------|---------|:--------:|
+| item\_cache\_ttl\_milli\_second | Time after which item cache will invalidate. Default 5 minutes | `number` | `300000` | no |
+| name | n/a | `string` | n/a | yes |
+| node\_type | n/a | `string` | `"dax.t2.small"` | no |
+| query\_cache\_ttl\_\_milli\_second | Time after which item query will invalidate. Default 5 minutes | `number` | `300000` | no |
+| region | n/a | `string` | n/a | yes |
+| replication\_factor | n/a | `number` | `3` | no |
+| tables | n/a | `list(string)` | n/a | yes |
+| tags | n/a | `any` | n/a | yes |
+| vpc\_id | n/a | `string` | n/a | yes |
```
Comment (shubhamPeak) [inline dax/README.md:23]: can you add the description of the fields where `n/a` is specified

**Review by shubhamPeak — APPROVED** [review]
looks good to me, just few readme file updates

**Review by shreyance-jain — APPROVED** [review]

**Review by azhar22k — APPROVED** [review]
Some good comments from @shubhamPeak . rest looks good

File: dax/README.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    source  = "git::https://github.com/peak-ai/terraform-modules.git//tags?ref=v0.7.0"
+    tenant  = "new-client"
+    stage   = "latest"
+    feature = "example"
+    service = "example"
+}
+
+module "my_dax" {
+  source  = "git::https://github.com/peak-ai/terraform-modules.git//dax?ref=v0.7.0"
+  name    = "dax"
```
Comment (azhar22k) [inline dax/README.md]: this example is incorrect, there are multiple required inputs for this module. should specify all of them in example (with dummy values)

File: dax/main.tf [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  }
+}
+
+data "aws_caller_identity" "current" {}
+resource "aws_iam_role" "role" {
+  name = "${var.name}-dax"
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
```
Comment (DebjyotiRay) [inline dax/main.tf:14]: can we bump this?
it's too late now. it is 2026.

**Pull request merged**